src/functions/routers/types.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  FunctionContext,
+  FunctionDefinitionArgs,
+  FunctionParameters,
+  FunctionRuntimeParameters,
+} from "../types.ts";
+
+export type ActionHandler<Definition> = Definition extends
```

Comment:
nit - could we consider naming this `BlockActionHandler` (and a few other Action* types) for consistency?